rename_imports.py

```diff
@@ -0,0 +1,78 @@
+
+from LSP.plugin.core.file_watcher import FileWatcherEvent, FileWatcherProtocol, get_file_watcher_implementation, FileWatcher
+from LSP.plugin.core.typing import  List
+import os
+import sublime
+
+
+def register_rename_import_file_watcher(root_path: str):
+    file_watcher = get_file_watcher_implementation()
+    if file_watcher:
+        watcher = file_watcher.create(
+            root_path=root_path,
+            patterns=["**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"],
+            events=["create", "delete"],
+            ignores=['**/node_modules/**'],
+            handler=HandleRenameImport
+        )
```

Comment:
"create" returns a controller and it is responsibility of the callee to destroy() it when the server process is closing. Not doing that will result in the file watcher process running even after server has closed.

Reply:
Do you know a hook to listen when the session is shutdown?
I could not find such method. 

I tried adding a `notification_handler` and listen to `exit` , but it is not called.

```
    @notification_handler("exit")
    def on_close(self, x):
        print("shutdown called")
```

Reply:
`AbstractPlugin.on_session_end_async`